apps/wallet/src/background/keyring/index.ts

```diff
@@ -206,23 +305,67 @@ class Keyring {
     }
 
     private async revive() {
-        const unlocked = await this.#vault.revive();
+        const unlocked = await this.#vaultStorage.revive();
         if (unlocked) {
-            this.unlocked();
+            await this.unlocked();
         }
     }
 
-    private unlocked() {
-        let mnemonic = this.#vault.getMnemonic();
+    private async unlocked() {
+        let mnemonic = this.#vaultStorage.getMnemonic();
         if (!mnemonic) {
             return;
         }
         Alarms.setLockAlarm();
-        this.#keypair = Ed25519Keypair.deriveKeypair(mnemonic);
+        const lastAccountIndex = await this.getLastDerivedIndex();
+        for (let i = 0; i <= lastAccountIndex; i++) {
+            const account = this.deriveAccount(i, mnemonic);
+            this.#accountsMap.set(account.address, account);
+            if (i === 0) {
+                this.#mainDerivedAccount = account.address;
+            }
+        }
+        this.#vaultStorage.getImportedKeys()?.forEach((anImportedKey) => {
+            const account = new Account('imported', { keypair: anImportedKey });
+            this.#accountsMap.set(account.address, account);
+        });
         mnemonic = null;
         this.#locked = false;
         this.notifyLockedStatusUpdate(this.#locked);
     }
+
+    private deriveAccount(accountIndex: number, mnemonic: string) {
+        const derivationPath = this.makeDerivationPath(accountIndex);
+        const keypair = Ed25519Keypair.deriveKeypair(mnemonic, derivationPath);
+        return new Account('derived', { keypair, derivationPath });
+    }
+
+    private async getLastDerivedIndex() {
+        return (
+            await Browser.storage.local.get({
+                [STORAGE_LAST_ACCOUNT_INDEX_KEY]: 0,
+            })
+        )[STORAGE_LAST_ACCOUNT_INDEX_KEY] as number;
+    }
+
+    private storeLastDerivedIndex(index: number) {
+        return Browser.storage.local.set({
+            [STORAGE_LAST_ACCOUNT_INDEX_KEY]: index,
+        });
+    }
+
+    private storeActiveAccount(address: SuiAddress) {
+        return Browser.storage.local.set({ [STORAGE_ACTIVE_ACCOUNT]: address });
+    }
+
+    private makeDerivationPath(index: number) {
+        // currently returns only Ed25519 path
+        return `m/44'/784'/0'/0'/${index}'`;
```

Comment:
would recommend to keep it consistent with the blog https://tech.mystenlabs.com/cryptography-in-sui-wallet-specifications/
```
m/44'/784'/${index}'/0'/0'
```

this is also the latest ledger "standard" for ledger live https://gist.github.com/miguelmota/034c81886bd23b25c64ec5eda9251641#file-hdpaths-txt-L2

Reply:
Makes sense! I'll change it 👍